src/dialogs/more-info/controls/more-info-cover.ts

```diff
@@ -1,89 +1,195 @@
-import { css, CSSResult, html, LitElement, nothing } from "lit";
-import { customElement, property } from "lit/decorators";
-import { attributeClassNames } from "../../../common/entity/attribute_class_names";
+import { mdiMenu, mdiSwapVertical } from "@mdi/js";
 import {
-  FeatureClassNames,
-  featureClassNames,
-} from "../../../common/entity/feature_class_names";
+  css,
+  CSSResultGroup,
+  html,
+  LitElement,
+  nothing,
+  PropertyValues,
+} from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { computeStateDisplay } from "../../../common/entity/compute_state_display";
 import { supportsFeature } from "../../../common/entity/supports-feature";
+import { blankBeforePercent } from "../../../common/translations/blank_before_percent";
 import "../../../components/ha-attributes";
-import "../../../components/ha-cover-tilt-controls";
-import "../../../components/ha-labeled-slider";
-import {
-  CoverEntity,
-  CoverEntityFeature,
-  isTiltOnly,
-} from "../../../data/cover";
-import { HomeAssistant } from "../../../types";
-
-export const FEATURE_CLASS_NAMES: FeatureClassNames<CoverEntityFeature> = {
-  [CoverEntityFeature.SET_POSITION]: "has-set_position",
-  [CoverEntityFeature.OPEN_TILT]: "has-open_tilt",
-  [CoverEntityFeature.CLOSE_TILT]: "has-close_tilt",
-  [CoverEntityFeature.STOP_TILT]: "has-stop_tilt",
-  [CoverEntityFeature.SET_TILT_POSITION]: "has-set_tilt_position",
-};
+import { CoverEntity, CoverEntityFeature } from "../../../data/cover";
+import type { HomeAssistant } from "../../../types";
+import "../components/cover/ha-more-info-cover-toggle";
+import "../components/cover/ha-more-info-cover-buttons";
+import "../components/cover/ha-more-info-cover-position";
+import "../components/cover/ha-more-info-cover-tilt-position";
+import { moreInfoControlStyle } from "../components/ha-more-info-control-style";
+import "../components/ha-more-info-state-header";
 
 @customElement("more-info-cover")
 class MoreInfoCover extends LitElement {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
-  @property({ attribute: false }) public stateObj!: CoverEntity;
+  @property({ attribute: false }) public stateObj?: CoverEntity;
+
+  @state() private _displayedPosition?: number;
+
+  @state() private _mode?: "position" | "button";
+
+  private _toggleMode() {
+    this._mode = this._mode === "position" ? "button" : "position";
+  }
+
+  private _positionChanged(ev) {
+    const value = (ev.detail as any).value;
+    if (isNaN(value)) return;
+    this._displayedPosition = value;
+  }
+
+  protected updated(changedProps: PropertyValues): void {
```

Comment:
do this in `willUpdate` as it sets reactive properties, it will cause another update/render when done in updated (which is after render)